File: app/src/main/java/com/hiddenramblings/tagmo/amiibo/FlaskTag.java
package com.hiddenramblings.tagmo.amiibo;

import com.hiddenramblings.tagmo.charset.CharsetCompat;
import com.hiddenramblings.tagmo.nfctech.TagArray;

public class FlaskTag extends Amiibo {

    public FlaskTag(String[] name) {
        super(null, TagArray.bytesToLong(
                name[2].getBytes(CharsetCompat.UTF_8)
        ), name[0], null);
    }
}

File: app/src/main/java/com/hiddenramblings/tagmo/amiibo/FlaskTag.kt
package com.hiddenramblings.tagmo.amiibo

import com.hiddenramblings.tagmo.charset.CharsetCompat
import com.hiddenramblings.tagmo.nfctech.TagArray

class FlaskTag(name: List<String>) : Amiibo(null,
    TagArray.bytesToLong(name[1].toByteArray(CharsetCompat.UTF_8)), name[0], null
)